concrete/src/Entity/File/StorageLocation/StorageLocation.php

```diff
@@ -162,13 +167,15 @@ public function clearCache()
     public function delete()
     {
         $default = \Concrete\Core\File\StorageLocation\StorageLocation::getDefault();
-        $db = Database::get();
+        $app = Application::getFacadeApplication();
+        $db = $app->make('database');
 
-        $fIDs = $db->GetCol('select fID from Files where fslID = ?', [$this->getID()]);
+        // let's check if there is any file in this storage location and throw an exception if yes
+        $fIDs = $db->fetchAll('SELECT fID FROM Files WHERE fslID = ?', [$this->getID()]);
```

Comment:
I havent tested yet but im pretty sure this wont actually stop storages with files getting deleted.

FetchAll with use pdo::fetch_both by default which would be like
```
[
[0]=>[fID]=>1,
[1]=>[fID]=>2,
]
```

You have a hasFiles function now use it also, if the default storage has 0 files should you be able to delete it? Even though it has 0 files in the database it still contains all the cache/generated thumbnails/avatars.